middleware/base_rate_limiter.rb

```diff
@@ -0,0 +1,152 @@
+module CloudFoundry
+  module Middleware
+    RequestCount = Struct.new(:requests, :valid_until)
+
+    class BaseRequestCounter
+      include CloudFoundry::Middleware::UserResetInterval
+
+      def initialize
+        reset
+      end
+
+      # needed for testing
+      def reset
```

Comment:
This can be made private; in the test you can use `request_counter.send(:reset)` (it seems this has already been done in other places in the codebase).